vulture_os/toolkit/api_parser/vectra/vectra.py

```diff
@@ -209,20 +199,53 @@ def test(self):
     def execute(self):
         try:
             ## GET LOGS ##
-            since = self.frontend.last_api_call or (timezone.now() - timedelta(days=1))
+            since = self.frontend.last_api_call or (timezone.now() - timedelta(days=2))
             to = min(timezone.now(), since + timedelta(hours=24))
 
             logger.info(f"[{__parser__}]:execute: getting logs from {since} to {to}", extra={'frontend': str(self.frontend)})
-            results = self.get_logs(since, to)
+            kinds = ["detections"]
+            results = list()
+            logs = list()
+            for kind in kinds:
+                since = self.last_collected_timestamps.get(f"vectra_{kind}") or self.frontend.last_api_call or (timezone.now() - timedelta(days=1))
+                results = self.get_logs(since, to, kind)
+                if results:
+                    new_timestamp = datetime.fromisoformat(results[-1]["event_timestamp"].replace("Z", "+00:00")) + timedelta(seconds=1)
+                    logger.info(f"[{__parser__}][execute] :: Successfully gets {len(results)} '{kind}' logs, updating last_collected_timestamps['vectra_{kind}'] -> {new_timestamp}", extra={'frontend': str(self.frontend)})
+                    self.last_collected_timestamps[f"vectra_{kind}"] = new_timestamp
+                    logs.extend(results)
+                else:
+                    if to == since + timedelta(hours=24):
+                        self.last_collected_timestamps[f"vectra_{kind}"] = self.last_collected_timestamps[f"vectra_{kind}"] +timedelta(hours=1)
+
+            ## GET DATA FROM ENTITIES ##
+            types = [
+                "account",
+                "host"
+            ]
+            for type in types:
+                since = self.last_collected_timestamps.get(f"vectra_{type}") or self.frontend.last_api_call or (timezone.now() - timedelta(days=1))
+                results = self.get_data_from_entity_api(since, to, type)
+                if results:
+                    new_timestamp = datetime.fromisoformat(results[-1]["event_timestamp"].replace("Z", "+00:00")) + timedelta(seconds=1)
+                    logger.info(f"[{__parser__}][execute] :: Successfully gets {len(results)} '{kind}' logs, updating last_collected_timestamps['vectra_{kind}'] -> {new_timestamp}", extra={'frontend': str(self.frontend)})
+                    self.last_collected_timestamps[f"vectra_{type}"] = new_timestamp
+                    logs.extend(results)
+                else:
+                    if to == since + timedelta(hours=24):
+                        self.last_collected_timestamps[f"vectra_{type}"] = self.last_collected_timestamps[f"vectra_{type}"] +timedelta(hours=1)
```

Comment:
prefer keeping the same format than for other collectors:

```suggestion
                elif since < timezone.now() - timedelta(hours=24):
```